### ISSUE TITLE
`EARTHLY_TARGET_TAG_DOCKER` does not work in Earthly CI

### ISSUE THREAD

**Issue by vladaionescu** [earthly/cloud-issues#11]
Presumably, because the branch information is not passed along. Perhaps there's a way in which we can feed that through in Earthly Core as an override.

**Comment by r-LaForge** [comment]
I think this is working as documented in that we provide a canonical reference that is compatible with docker, it just happens to be a commit hash. It isn't explicitly called out that this should be a branch name.
 
```
The tag part of the canonical reference of the current target, sanitized for safe use as a docker tag. This is guaranteed to be a valid docker tag, even if no canonical form exists, in which case, latest is used.
```

In CI we check out a specific branch (because we're interested in running against a specific commit).

I just tested this and got the following:
<img width="963" alt="image" src="https://user-images.githubusercontent.com/13823057/224888075-e845d28b-23eb-4945-8a4d-1488a1fcf1c6.png">

Earthfile pipeline:
```
test-args-pipeline:
  PIPELINE
  TRIGGER push ryan/test-config-branch
  TRIGGER pr main
  BUILD +test-args

test-args:
  ARG EARTHLY_TARGET_TAG_DOCKER
  ARG EARTHLY_TARGET_TAG
  RUN echo $EARTHLY_TARGET_TAG_DOCKER 
  RUN echo $EARTHLY_TARGET_TAG 
```

**Comment by vladaionescu** [comment]
I guess so, although the intended behavior was that this would be the branch name if this is triggered on the activity of that branch. So yeah, technically it is not deviating from what the docs say, but also it's not what this was intended to be.

I believe the algorithm for its value should be:

* If this is triggered as a result of a tag, it should be that tag.
* If this is triggered on activity of a branch (push or pr trigger), it should be the branch.
* All other cases, the git sha.

**Comment by r-LaForge** [comment]
Ah... thanks for clearing that up.

**Comment by r-LaForge** [comment]
This may also be impacting the `EARTHLY_GIT_BRANCH` ARG

## LOCALLY
<img width="1462" alt="image" src="https://user-images.githubusercontent.com/13823057/227573098-c2a086e3-24e5-4e32-9e84-082f259784d2.png">

## CI
![image](https://user-images.githubusercontent.com/13823057/227573915-00a5dbcd-a7d4-486c-850f-98276624da8e.png)

```
deploy:
    FROM google/cloud-sdk:423.0.0-alpine
    RUN apk add go-task-task
    ARG EARTHLY_GIT_BRANCH
    COPY Taskfile.yml ./
    RUN --no-cache echo Earthly git branch is ${EARTHLY_GIT_BRANCH}
    IF --no-cache export GIT_BRANCH=$EARTHLY_GIT_BRANCH && echo $GIT_BRANCH && task should-deploy
        RUN echo $EARTHLY_GIT_BRANCH
        RUN --secret GCP_KEY=gke-deploy echo ${GCP_KEY} > /tmp/gke-deploy.json && \
                        gcloud auth activate-service-account --key-file /tmp/gke-deploy.json
    END

deploy-pipeline:
    PIPELINE
    TRIGGER manual
    TRIGGER push develop
    TRIGGER pr develop
    TRIGGER pr main
    TRIGGER push main
    #BUILD +all
    BUILD +deploy
```

**Comment by r-LaForge** [comment]
It was solved on the earthly Core side - but I still need to do a follow-up PR in cloud.
I'll update this issue once that is done.

**Comment by r-LaForge** [comment]
This issue has now been resolved in CI.

$EARTHLY_GIT_BRANCH will be the branch name; whereas, $EARTHLY_TARGET_TAG_DOCKER will be the branch name, but adjusted to be appropriate for use as a docker tag.
